### PR TITLE
[Features] Registration / authorization / memorization

### PR DESCRIPTION
**Pull request by B0RlS**
**Link to the ticket :** 
https://trello.com/c/c6thC9Lu/7-feature-user-registration-authorization-memorization
**What it does:**
1) Created users and sessions controllers
2) Created user model
3) Created session helper
4) Created a main page with a header for convenience.
5) There is a login button for unauthorized users
6) After clicking on the button, the user can log in if he has an account or be registered
7) A system of hashing and hiding information from data leakage was introduced.
8) Bcrypt was used
9) The function of saving the session when closing the browser has been added, at the entrance you can specify "remember me"
10) Added validation of input information
11) Possible options were thought out and implemented: opening a page from different browsers, opening several tabs, logging out of an account with open tabs, user session status
12) Tested the work of controllers, helpers and models
**DB changes:**
Created table users.
**UI changes:**
Created views for HomeController
Created views for registration and authorization
Created views for UsersController

### CODE REVIEW

File: app/assets/stylesheets/home.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Home controller here.
```
Comment (labs-ruby) [inline app/assets/stylesheets/home.scss]: Remove empty scss and coffee files if not needed

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (labs-ruby) [inline app/helpers/users_helper.rb]: Remove helpers if not needed

File: spec/controllers/sessions_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  describe 'DELETE #destroy' do
+    it 'redirect to root' do
+      delete :destroy
+      expect(response).to redirect_to(root_url)
+    end
+  end
+
+  # describe 'POST #create' do
```
Comment (labs-ruby) [inline spec/controllers/sessions_controller_spec.rb]: Why tests are commented?

File: spec/views/sessions/new.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (labs-ruby) [inline spec/views/sessions/new.html.erb_spec.rb]: Empty specs!

File: spec/helpers/sessions_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+# Specs in this file have access to a helper object that includes
```
Comment (labs-ruby) [inline spec/helpers/sessions_helper_spec.rb]: Please remove useless comments!

**Comment by B0RlS** [comment]
[Fix all rubocop errors in project](https://trello.com/c/DADUAsZG/65-fix-all-rubocop-errors-in-project)

**Pull request merged**